starter-code/pizza.js

```diff
@@ -1,85 +1,161 @@
+};
 
 // Initial value of the state (the state values can change over time)
 var state = {
   pepperonni: true,
   mushrooms: true,
   greenPeppers: true,
   whiteSauce: false,
-  glutenFreeCrust: false
-}
+  glutenFreeCrust: false,
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll('.pep').forEach(function($pep) {
     if (state.pepperonni) {
-      $pep.style.visibility = "visible";
-    }
-    else {
-      $pep.style.visibility = "hidden";
+      $pep.style.visibility = 'visible';
+    } else {
+      $pep.style.visibility = 'hidden';
     }
-  })
+  });
 }
 
 function renderMushrooms() {
+  console.log('functionCalled');
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = 'visible';
+    } else {
+      $mush.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function($greenP) {
+    if (state.greenPeppers) {
+      $greenP.style.visibility = 'visible';
+    } else {
+      $greenP.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce-white').forEach(function($sauce) {
+    if (state.whiteSauce) {
+      $sauce.style.visibility = 'visible';
+    } else {
+      $sauce.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll('.crust-gluten-free').forEach(function($gluten) {
+    if (state.glutenFreeCrust) {
+      $gluten.style.visibility = 'visible';
+    } else {
+      $gluten.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+  return document.querySelectorAll('.btn').forEach(function(element, index) {
+    let keyArr = Object.keys(state); // ["pepperonni", "mushrooms","greenPeppers", "whiteSauce", "glutenFreeCrust"]
+
+    if (state[keyArr[index]]) {
+      element.classList.add('active');
+    } else {
+      element.classList.remove('active');
```

Comment:
For the buttons you could also have use the toggle function for each button html element in the onclick functions
Something like this:
```
document.querySelector(".btn.btn-pepperonni").onclick = function() {
  event.target.classList.toggle("active");
  state.pepperonni = !state.pepperonni;
  renderEverything();
};

```

Both solutions are good though :)